apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,563 @@
+      "default": "./dist/cjs/index.js"
+    }
+  }
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-with-wrapper-dual-distro",
+  "type": "commonjs",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": {
+      "import": "./dist/esm/wrapper.mjs",
+      "require": "./dist/cjs/index.js",
+      "default": "./dist/cjs/index.js"
+    },
+    "./package.json": "./package.json"
+  }
+}
+```
+
+In order to support named exports from the CJS bundle for an ESM consumer, this will need a bit of gymnastics from a bundler but is conceptually very simple.
+
+In certain conditions, CJS exports an object (which gets aliased to ESM's `default`); that object, like any object, is destructure-able. You can leverage that to pluck all the members of the object out, and then re-export them so the ESM consumer is none the wiser.
+
+```js displayName="./dist/esm/wrapper.mjs"
+import cjs from '../cjs/index.js';
+
+const { a, b, c /* … */ } = cjs;
+
+export { a, b, c /* … */ };
+```
+
+#### Two full distributions
+
+The "Long Island Ice Tea" of packages: Chuck in a bunch of stuff and hope for the best. This is probably the most common and easiest of the CJS to CJS & ESM options, but you pay for it. This is rarely a good idea.
+
+Pros:
+
+- Simple bundler configuration
+
+Cons:
+
+- Larger package weight (basically double)
+- Vulnerable to the Dual-Package Hazard
+
+**Working example**: [cjs-with-dual-distro (double)](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/dual/double-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-with-full-dual-distro",
+  "exports": {
+    ".": {
+      "import": "./dist/esm/index.mjs",
+      "require": "./dist/cjs/index.js",
+      "default": "./dist/cjs/index.js"
+    }
+  }
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-with-full-dual-distro",
+  "type": "commonjs",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": {
+      "import": "./dist/esm/index.mjs",
+      "require": "./dist/cjs/index.js",
+      "default": "./dist/cjs/index.js"
+    },
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Alternatively, you can use `"default"` and `"node"` keys, which are less counter-intuitive: Node.js will always choose the `"node"` option (which always works), and non-Node.js tooling will choose `"default"` when configured to target something other than node. **This precludes the dual-package hazard.**
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-with-alt-full-dual-distro",
+  "exports": {
+    ".": {
+      "default": "./dist/esm/index.mjs",
+      "node": "./dist/cjs/index.js"
+    }
+  }
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-with-alt-full-dual-distro",
+  "type": "commonjs",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": {
+      "default": "./dist/esm/index.mjs",
```

Comment:
I am pretty sure it's incorrect to put default before any other exports?

Reply:
OH! You're right! But also, how and why does this possibly matter 🤔

Reply:
For this specific case, the dual module doesn't work on versions of Node.js that doesn't support `require(esm)`, because default gets picked first.